core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -151,4 +158,61 @@ public static DataMapStoreManager getInstance() {
     return instance;
   }
 
+  public TableSegmentRefresher getTableSegmentRefresher(AbsoluteTableIdentifier identifier) {
+    String uniqueName = identifier.uniqueName();
+    if (segmentRefreshMap.get(uniqueName) == null) {
+      segmentRefreshMap.put(uniqueName, new TableSegmentRefresher(identifier));
+    }
+    return segmentRefreshMap.get(uniqueName);
+  }
+
+  /**
+   * Keep track of the segment refresh time.
+   */
+  public static class TableSegmentRefresher {
+
+    private Map<String, Long> segmentRefreshTime = new HashMap<>();
```

Comment:
add comment to describe the content, like map segmentId to last updated timestamp of the segment

Reply:
ok